db/node_jobs.go

```diff
@@ -193,111 +243,172 @@ func UpdateBLSToExecutionChangesNodeJob(job *types.BLSToExecutionChangesNodeJob)
 func SubmitBLSToExecutionChangesNodeJobs() error {
 	maxSubmittedJobs := 100
 	jobs := []*types.NodeJob{}
-	err := WriterDb.Select(&jobs, `select id, type, status, created_time, submitted_to_node_time, completed_time, data from node_jobs where type = $1 and status = $2 limit $4-(select count(*) from node_jobs where type = $1 and status = $3)`, types.BLSToExecutionChangesNodeJobType, types.PendingNodeJobStatus, types.SubmittedToNodeNodeJobStatus, maxSubmittedJobs)
+	err := WriterDb.Select(&jobs, `select id, type, status, created_time, submitted_to_node_time, completed_time, data from node_jobs where type = $1 and status = $2 order by created_time limit $4-(select count(*) from node_jobs where type = $1 and status = $3)`, types.BLSToExecutionChangesNodeJobType, types.PendingNodeJobStatus, types.SubmittedToNodeNodeJobStatus, maxSubmittedJobs)
 	if err != nil {
 		return err
 	}
 	for _, job := range jobs {
-		j, err := job.ToBLSToExecutionChangesNodeJob()
+		err = job.ParseData()
 		if err != nil {
 			return err
 		}
-		err = SubmitBLSToExecutionChangesNodeJob(j)
+		err = SubmitBLSToExecutionChangesNodeJob(job)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func SubmitBLSToExecutionChangesNodeJob(job *types.BLSToExecutionChangesNodeJob) error {
-	data, err := json.Marshal(job.Data)
-	if err != nil {
-		return err
-	}
+func SubmitBLSToExecutionChangesNodeJob(job *types.NodeJob) error {
 	client := &http.Client{Timeout: time.Second * 10}
 	url := fmt.Sprintf("%s/eth/v1/beacon/pool/bls_to_execution_changes", utils.Config.NodeJobsProcessor.ClEndpoint)
-	resp, err := client.Post(url, "application/json", bytes.NewReader(data))
+	resp, err := client.Post(url, "application/json", bytes.NewReader(job.RawData))
 	if err != nil {
 		return err
 	}
 	if resp.StatusCode != 200 {
 		d, _ := ioutil.ReadAll(resp.Body)
-		return fmt.Errorf("http request error: %s: %s, data: %s", resp.Status, d, data)
+		return fmt.Errorf("http request error: %s: %s, data: %s", resp.Status, d, job.RawData)
 	}
-	_, err = WriterDb.Exec(`update node_jobs set status = $1 where id = $2`, types.SubmittedToNodeNodeJobStatus, job.ID)
+	job.Status = types.SubmittedToNodeNodeJobStatus
```

Comment:
**Minor:** This is duplicated code, see line 406. You should move it into a function, something like `submitJobInDb`.